starboard/nplb/posix_compliance/posix_aligned_alloc_test.cc

```diff
@@ -0,0 +1,146 @@
+  // Valid alignment is a power of 2 based on
+  // https://man7.org/linux/man-pages/man3/aligned_alloc.3.html#DESCRIPTION
+  const size_t kValidAlignment = 64;
+  const size_t kValidSize = kValidAlignment * 2;
+  ASSERT_TRUE((kValidAlignment > 0) &&
+              ((kValidAlignment & (kValidAlignment - 1)) == 0));
+  ASSERT_EQ(0, kValidSize % kValidAlignment);
+
+  errno = 0;
+  void* ptr = aligned_alloc(kValidAlignment, kValidSize);
+
+  ASSERT_NE(nullptr, ptr) << "aligned_alloc returned NULL. errno: " << errno;
+  EXPECT_EQ(0, errno);
+  EXPECT_TRUE(IsMemoryAligned(ptr, kValidAlignment));
+
+  free(ptr);
+}
+
+TEST(PosixAlignedAllocTests, HandlesZeroSizeCorrectly) {
+  const size_t kValidAlignment = 128;
+  const size_t kZeroSize = kValidAlignment * 0;
+  ASSERT_TRUE((kValidAlignment > 0) &&
+              ((kValidAlignment & (kValidAlignment - 1)) == 0));
+
+  errno = 0;
+  void* ptr = aligned_alloc(kValidAlignment, kZeroSize);
+
+  // For size 0, aligned_alloc may return NULL or a unique pointer.
+  // It should not set errno to EINVAL if alignment is valid.
+  EXPECT_NE(EINVAL, errno) << "errno was " << errno;
+
+  if (ptr != nullptr) {
+    EXPECT_TRUE(IsMemoryAligned(ptr, kValidAlignment));
+  }
+  free(ptr);
+}
+
+TEST(PosixAlignedAllocTests,
+     AllocatesSuccessfullyWithMinimumValidAlignmentOne) {
+  const size_t kAlignmentOne = 1;  // 2^0 = 1
+  const size_t kAnySize = kAlignmentOne * 100;
+
+  errno = 0;
+  void* ptr = aligned_alloc(kAlignmentOne, kAnySize);
+
+  ASSERT_NE(nullptr, ptr) << "aligned_alloc returned NULL. errno: " << errno;
+  EXPECT_EQ(0, errno);
+  EXPECT_TRUE(IsMemoryAligned(ptr, kAlignmentOne));
+
+  free(ptr);
+}
+
+TEST(PosixAlignedAllocTests,
+     AllocatesSuccessfullyWithAlignmentEqualToSizeofVoidPointer) {
+  const size_t kAlignmentSizeofVoidPtr = sizeof(void*);
+  if (!((kAlignmentSizeofVoidPtr > 0) &&
+        ((kAlignmentSizeofVoidPtr & (kAlignmentSizeofVoidPtr - 1)) == 0))) {
+    GTEST_SKIP() << "sizeof(void*) (" << kAlignmentSizeofVoidPtr
+                 << ") is not a power of two on this system, skipping test.";
+  }
+  const size_t kTestSize = kAlignmentSizeofVoidPtr * 4;
+
+  errno = 0;
+  void* ptr = aligned_alloc(kAlignmentSizeofVoidPtr, kTestSize);
+
+  ASSERT_NE(nullptr, ptr) << "aligned_alloc returned NULL. errno: " << errno;
+  EXPECT_EQ(0, errno);
+  EXPECT_TRUE(IsMemoryAligned(ptr, kAlignmentSizeofVoidPtr));
+
+  free(ptr);
+}
+
+TEST(PosixAlignedAllocTests, AllocatesSuccessfullyWithLargeAlignmentAndSize) {
+  const size_t kLargeAlignment = 1024 * 1024;
+  const size_t kLargeSize = kLargeAlignment * 2;
+  ASSERT_TRUE((kLargeAlignment > 0) &&
+              ((kLargeAlignment & (kLargeAlignment - 1)) == 0));
+  ASSERT_EQ(0, kLargeSize % kLargeAlignment);
+
+  errno = 0;
+  void* ptr = aligned_alloc(kLargeAlignment, kLargeSize);
+
+  if (ptr == nullptr) {
+    // ENOMEM is a possible outcome for large allocations, but not EINVAL.
+    EXPECT_NE(EINVAL, errno)
```

Comment:
It seems like this whole block could be converted to a `EXPECT_EQ(ENOMEM, errno)`